### PR TITLE
Feat/api tags filter

### PR DESCRIPTION
**Pull request by santokhsingh**

### CODE REVIEW

File: routes/apis.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 var express = require('express');
+var where = require("lodash");
```
Comment (DonMartin76) [inline routes/apis.js]: No need to pull in `lodash` for `filter`; in node.js, `Array.prototype.filter()` is already defined. Plus you never added `lodash` to `package.json` (and please don't).

File: routes/apis.js [hunk 2]
```diff
@@ -26,28 +27,57 @@ router.get('/', function (req, res, next) {
                 return next(err);
             var apiList = results.getApis.apis;
             // Markdownify short descriptions.
+            var apiTags = [];
             for (var i = 0; i < apiList.length; ++i) {
                 if (apiList[i].desc)
                     apiList[i].desc = marked(apiList[i].desc);
+                if(apiList[i].tags.length>0){
```
Comment (DonMartin76) [inline routes/apis.js]: This will break if `tags` is not defined. Please check for existence first.

File: routes/apis.js [hunk 3]
```diff
@@ -26,28 +27,57 @@ router.get('/', function (req, res, next) {
...
+                  for (var j = 0; j < apiList[i].tags.length; ++j) {
+                    apiTags.push(apiList[i].tags[j]);
+                  }
+                }
+            }
+            if(req.query["filter"]){
+              apiList=where.filter(apiList, function(api) {
```
Comment (DonMartin76) [inline routes/apis.js]: `apiList.filter(<predicate function>)`

**Comment by DonMartin76** [comment]
I will have a couple of remarks on this, but I want to test it first.

**Comment by santokhsingh** [comment]
Implemented all your suggestion. I must admit, it was little average quality PR on my part, will be more careful in the future

**Pull request merged**